add ecommerce to rich messages (#1656)
livechat/livechat-public-docs: src/pages/messaging/agent-chat-api/data-structures/index.mdx

| `postback`                      | no       | `object`  | Indicates that the message event was generated in response to a rich message event.                                          |
| `postback.id`                   | yes      | `string`  | ID of the postback from the rich message event.                                                                              |
| `postback.thread_id`            | yes      | `string`  | ID of the thread with the rich message event.                                                                                |
| `postback.event_id`             | yes      | `string`  | ID of the rich message event.                                                                                                |
| `postback.type`                 | no       | `string`  | Should be used together with `postback.value` (when  one of them is present, the other is required).                         |
| `postback.value`                | no       | `string`  | Should be used together with `postback.type` (when  one of them is present, the other is required).                          |

> | `postback.ecommerce`            | no       | `object`  | Object containing data specific to rich messages with `ecommerce` `template_id` like `product_id`.                           |

Reviewer: ```suggestion
| `postback.ecommerce`            | no       | `object`  | Object containing data specific to rich messages with `ecommerce` template like `product_id`.                           |
```
opt

> | `postback.ecommerce`            | no       | `object`  | Object containing data specific to rich messages with `ecommerce` template like `product_id`.                                |

Reviewer 2: ```suggestion
| `postback.ecommerce`            | no       | `object`  | Object containing data specific to rich messages with `ecommerce` template, like `product_id`.                                |
```

> | `postback.ecommerce.product_id` | yes      | `string`  | ID of the product displayed by the rich message.                                                                             |

Reviewer 2: ```suggestion
| `postback.ecommerce.product_id` | no      | `string`  | The ID of the product displayed by the rich message. Required when sending `postback.ecommerce`.                                                                           |
```

> | `postback.ecommerce.option_id`  | yes      | `string`  | ID of the selected option for the product.                                                                                   |

Reviewer 2: ```suggestion
| `postback.ecommerce.option_id`  | no      | `string`  | ID of the selected option for the product. Required when sending `postback.ecommerce`.                                                                                   |
```

| `text`                          | always     |                                                                                                 |
| `postback`                      | optionally | Appears in a message only when triggered by a rich message.                                     |
| `postback.id`                   | always     |                                                                                                 |
| `postback.thread_id`            | always     |                                                                                                 |
| `postback.event_id`             | always     |                                                                                                 |
| `postback.type`                 | optionally | Appears only if `postback.value` is present.                                                    |
| `postback.value`                | optionally | Appears only if `postback.type` is present.                                                     |
| `postback.ecommerce`            | optionally |                                                                                                 |

> | `postback.ecommerce.product_id` | always     |                                                                                                 |

Reviewer 2: ```suggestion
| `postback.ecommerce.product_id` | optionally     |                                                                                                 |
```

> | `postback.ecommerce.option_id`  | always     |                                                                                                 |

Reviewer 2: ```suggestion
| `postback.ecommerce.option_id`  | optionally     |                                                                                                 |
```

| `elements.buttons.user_ids`                      | yes      | `array`   | Describes users that sent the postback with `"toggled": true`.                                                                                                                                                                                   |
| `elements.buttons.target`                        | no       | `string`  | Should be used for the `url` button `type`. Specifies if the URL should open in the current or a new tab. Possible values: `new`, `current`.                                                                                                     |
| `elements.ecommerce`                             | no       | `object`  | Should be used for the `ecommerce` template. Contains information about displayed product and its variants.                                                                                                                                      |

> | `elements.ecommerce.view_type`                   | yes      | `string`  | Defines how the product and its options are displayed. Possible values: `tags`, `swatch`, `images`, `select`.                                                                                                                                    |

Reviewer 2: ```suggestion
| `elements.ecommerce.view_type`                   | no      | `string`  | Defines how the product and its options are displayed. Possible values: `tags`, `swatch`, `images`, `select`. Required when sending `elements.ecommerce`.                                                                                                                                    |
```

> | `elements.ecommerce.product_id`                  | yes      | `string`  |                                                                                                                                                                                                                                                  |

Reviewer 2: ```suggestion
| `elements.ecommerce.product_id`                  | no      | `string`  | Required when sending `elements.ecommerce`.                                                                                                                                                                                                                                                 |
```

| `elements.buttons.webview_height` | always     | Unless button `type` is different than `webview`.                                                                                                                       |
| `elements.buttons.postback_id`    | always     |                                                                                                                                                                         |
| `elements.buttons.user_ids`       | always     |                                                                                                                                                                         |
| `elements.buttons.target`         | optionally |                                                                                                                                                                         |
| `elements.ecommerce`              | optionally | This element is only present for rich messages with `template_id` set to `ecommerce`.                                                                                          |

> | `elements.ecommerce.view_type`    | always     | Can be either `tags`, `swatch`, `images` or `select`.                                                                                                                   |

Reviewer 2: ```suggestion
| `elements.ecommerce.view_type`    | optionally     | Can be either `tags`, `swatch`, `images` or `select`.                                                                                                                   |
```

> | `elements.ecommerce.product_id`   | always     |                                                                                                                                                                         |

Reviewer 2: ```suggestion
| `elements.ecommerce.product_id`   | optionally     |                                                                                                                                                                         |
```
